interoperator/controllers/provisioners/sfserviceinstancecleaner/sfserviceinstancecleaner_controller.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 The Service Fabrik Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	osbv1alpha1 "github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/api/osb/v1alpha1"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/constants"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/utils"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+)
+
+// SfServiceInstanceCleanerReconciler reconciles a SfServiceInstanceCleaner object
+type SfServiceInstanceCleanerReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+func (r *SfServiceInstanceCleanerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("sfserviceinstancecleaner", req.NamespacedName)
+	binding := &osbv1alpha1.SFServiceBinding{}
+	err := r.Get(ctx, req.NamespacedName, binding)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			// Object not found, return. Created objects are automatically
+			// garbage collected.
+			log.Info("binding deleted", "binding", req.NamespacedName.Name)
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err  // error reading object, requeue the request
+	}
+	if !binding.GetDeletionTimestamp().IsZero() {
+		finalizers := binding.GetFinalizers()
+		if utils.ContainsString(finalizers, constants.BrokerFinalizer) {
```

Comment:
why is this broker finalizer check required? Even if broker finalizer is not there, for a SFServicebinding which do not have corresponding SFServiceInstance, shouldn't we delete it ?

Reply:
The binding controller triggers delete for orphan bindings currently. So interoperator finalizer will be deleted by binding controller itself. The logic should be if interoperator finalizer is not present, set finalizer array to empty.

Reply:
yes, I was thinking if we can cover that part here as well, in that case there is no special handling the the controller itself, and this guy does all the reaping. In that case, binding and instance controllers would be similar(I guess we had discussed this earlier). I am fine not doing it in this PR as well. Whatever you feel better. @vivekzhere

Reply:
@subhankarc this is the updated workflow as suggested by @vivekzhere -
- check if the deletion timestamp is set
- check if the corresponding `SFServiceInstance` is absent

If the above two conditions are satisfied, patch the binding with an empty slice of finalizers.

Reply:
We will remove it from binding controller in separate PR

Reply:
@shivansh : Fair enough. This is what I was thinking, we do not need additional finalizer check. Would you please change the PR description accordingly?